modPrecompiled/src/org/aion/precompiled/ContractFactory.java

```diff
@@ -56,7 +53,8 @@ public ContractFactory(){}
      * @param track The repo.
      * @return the specified pre-compiled address.
      */
-    public static IPrecompiledContract getPrecompiledContract(ExecutionContext context,
+    @Override
+    public IPrecompiledContract getPrecompiledContract(ExecutionContext context,
         IRepositoryCache<AccountState, IDataWord, IBlockStoreBase <?, ?>> track) {
 
         switch (context.address().toString()) {
```

Comment:
The `context.address()` will be the contract address, rather than destination address, in case of `DELEGATECALL` and `CALLCODE`.

Reply:
Would that be intended behaviour? Unless we want to implement library precompiled contracts in the future...

Reply:
Good point. If we're not going to implement precompiled library in the future, then should be fine; otherwise, a hardfork would be required to enable it.

Reply:
I'm going to say this is intended behaviour for now.